typescript/web/src/components/workspace-switcher/create-workspace-modal/create-workspace-modal.tsx

```diff
@@ -101,8 +103,13 @@ export const useCreateWorkspace = (): [
   string | undefined
 ] => {
   const { name } = useWorkspaceNameInput();
+  const [plan] = useQueryParam("plan", StringParam);
+  const nonUndefinedPlan = plan ?? "";
+  const planToCreate = !(nonUndefinedPlan in WorkspacePlan)
+    ? DEFAULT_WORKSPACE_PLAN
```

Comment:
We should throw an error if the workspacePlan given in query params doesn't matches our own list.